### ISSUE TITLE
`Plugin Container for Firefox has stopped working` after driver.quit()

### ISSUE THREAD

**Issue by komlevv** [SeleniumHQ/selenium#3493]
Windows 7
Python 27
Selenium 3.0.2
Firefox 49
example code:
```
from selenium import webdriver
from selenium.webdriver.firefox.firefox_binary import FirefoxBinary
import os


# check gecko is in PATH
gecko_path = 'C:\\Program Files\\Geckodriver\\'
if 'Geckodriver' not in os.environ['PATH']:
    os.environ['PATH'] += os.pathsep + gecko_path  

profile = webdriver.FirefoxProfile()
firefox_path = 'C:\\Program Files\\Mozilla Firefox\\firefox.exe'
binary = FirefoxBinary(firefox_path)

driver = webdriver.Firefox(firefox_binary=binary, firefox_profile=profile)
driver.get("http://www.python.org")
driver.quit()
```
`Plugin Container for Firefox has stopped working`
get this error from `werfault.exe` after calling .quit() on driver
renaming or deleting plugin-container hangs Firefox upon next execution
fault module name: `mozglue.dll`

tested also on Firefox 50, 51
which doesn't have Plugin Container error message, 
but crash nonetheless with same fault module name.

driver.close() not working

also tried killing manually by pid, but can't get pid of firefox, since driver.binary doesn't seem to have `.process`:
`driver.binary.process` throws error `AttributeError: 'FirefoxBinary' object has no attribute 'process'`
`driver.binary.profile` throws error `AttributeError: 'FirefoxBinary' object has no attribute 'profile'`

**Comment by cgoldberg** [comment]
can you post the stacktrace that is produced if you run in a terminal?

**Comment by johny-bravo** [comment]
@cgoldberg 
driver.quit() exists without error in python, but i get plugin-container crashing

full error when i try to access driver.binary.process:
```
Traceback (most recent call last):
  File "H:/IdeaProjects/gmx_robot/tests/selen_err.py", line 39, in <module>
    print driver.binary.process
AttributeError: 'FirefoxBinary' object has no attribute 'process'

Process finished with exit code 1
```

`appcrash.exe` info for `plugin-container` when calling `driver.quit()`:
https://gist.github.com/johny-bravo/4959d71265260925a318afaa8dcb8f34

call trace for `driver.quit()`:
https://gist.github.com/johny-bravo/b4888e2256f1303011f7bba621af63d1

call trace for `driver.binary.process` error
https://gist.github.com/johny-bravo/a1a4f9246239bb2f321a9582d6208b09

**Comment by cgoldberg** [comment]
AFAIK plugin-container is a seperate process spawned by Firefox to load plugin content such as Flash.  So this looks to be an issue with Firefox/geckodriver, rather than selenium.

But most importantly, for anyone to debug this issue, you *need* to provide a minimal but complete test that shows this happen.  Please post the code you call selenium with (with everything not directly related to the issue stripped out). 


> print driver.binary.process
> AttributeError: 'FirefoxBinary' object has no attribute 'process'

that's pretty self explanatory... driver.binary simply doesn't have a process attribute exposed, so that won't work.I don't think selenium offers the underlying PID's of geckodriver and Firefox it is running.  Although you could always kill those processes externally with os commands in your test scripts.

**Comment by johny-bravo** [comment]
i can't kill a process without pid since i have multiple processes running and don't know which one to kill.
i could query windows task manager and diff before/after launch but it gets too complicated when multiple instances launch, since Popen is already in the code, it would be natural to just use driver.binary.process

driver.binary should have `process` initiated and exposed by the looks of it in
https://github.com/SeleniumHQ/selenium/blob/33d7d1a0e07100945b81cd88bb17a354887701e3/py/selenium/webdriver/firefox/firefox_binary.py#L66
for example `driver.service.process` works OK but it's for marionette/gecko and not firefox..

the code i use to reproduce is in the 1st post
the solution i use now is to kill werfault.exe but it's not ideal

**Comment by robinsmathewgit** [comment]
@johny-bravo 

I am also facing the same issue with

Windows 7
Java 1.8
Selenium 3.0.1
Firefox 51

Browser is getting launched but after that it crash with popup "_plugin container for firefox has stopped working selenium_" 

There were some posts asking to update the flash plugin to resolve the issue. I tried that, it never worked.

**Comment by Jan4them** [comment]
Windows 7 / Windows 8, 
Selenium 3.0.1 + geckodriver 0.14
Firefox 47 / Firefox 51

Below  a solution I found to eradicate the alert message about the 'Plugin Container for Firefox has stopped working' when quitting the driver / driver.quit() 

I notice that issue occurs when "killing" the Firefox process when open on an url containing flash, pdf  or other required plugins ... this can be reproduced at any time  - independently from selenium/geckodriver
ex. in python:  psutil.process( ffox-pid ).kill()

To solve it, it's key to send a "smooth ending request" to Firefox, such as ctrl+w and then to "kill" the geckodriver process, otherwise it causes other kind of problems having broken the connection .

To summarize : Smoothly close Firefox and then 'Kill' the Geckodriver process

in Python 
#Smoothly close firefox
driver.find_element_by_tag_name('body').send_keys(Keys.CONTROL, "w")
#Brutal end of geckodriver
for p in psutil.process_iter():
            if p.name() == 'geckodriver.exe':
                p.kill()
                break
#make some temp profile cleaning in C:\users\< user >\app.data\local\temp
...

**Comment by barancev** [comment]
This is a browser issue, we can't fix it on Selenium side. As a workaround one can disable "crash dialogues" as described here: https://www.raymond.cc/blog/disable-program-has-stopped-working-error-dialog-in-windows-server-2008/
